Model/ProductProtection.php

```diff
@@ -394,30 +430,88 @@ public function upsert(
     private function createOptions($product, $item, $updatedOptions): array
     {
         $options = [];
-        $optionIds = [];
-        foreach ($product->getOptions() as $o) {
-            $optionId = $o->getId();
-            $existingOption = $item->getOptionByCode("option_$optionId");
+        foreach (self::CUSTOM_OPTION_CODES as $optionCode) {
+            $existingOption = $item->getOptionByCode($optionCode);
             // if the option is in the updated options/values, create a new option
             // if there is no update to the option, use the existing option
-            if (isset($updatedOptions[$o->getTitle()])) {
+            if (isset($updatedOptions[$optionCode])) {
                 $option = $this->itemOptionFactory->create();
                 $option->setProduct($product);
-                $option->setCode("option_$optionId");
-                $option->setValue($updatedOptions[$o->getTitle()]);
+                $option->setCode($optionCode);
+                $option->setValue($updatedOptions[$optionCode]);
                 $options[] = $option;
```

Comment:
What I've realized is that you can create options without having them pre-defined on the product (i.e. our PP product) - I went this route as it seemed to simplify things and also stemmed away from using labels as a matching mechanism.

If we're happy with this, a follow up might be to investigate into removing the product options from the ProductInstaller since those may be ineffective now.